src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentSnapshot.cs

```diff
@@ -41,5 +41,16 @@ internal abstract class DocumentSnapshot
         public abstract bool TryGetGeneratedCSharpOutputVersion(out VersionStamp result);
 
         public abstract bool TryGetGeneratedHtmlOutputVersion(out VersionStamp result);
+
+        public async Task<VersionStamp> GetSemanticVersionAsync()
+        {
+            // In order to react to changing parameters or a file being deleted we need to consider the version of the CSharp backing document as well.
+            // Ideally we would operate based on the SemanticVersion of the CSharp document, but we don't currently have a way to compute that.
+            // Even if we did there's a strong chance that it would be slower than re-coloring on every C# change because it means a re-parse of the C# SyntaxTree.
+            var cSharpVersion = await GetGeneratedCSharpOutputVersionAsync();
```

Comment:
This wont be sufficient. We need to represent the ProjectSnapshot version in this DocumentSnapshot version because the ProjectSnapshot has all applicable TagHelpers which may or may not influence the generated output but would influence the semantic coloring/IntelliSense etc.. 

For instance if someone changes their TagHelper to also associate itself with an element name in addition to an attribute name the semantic tokens will never update because the generated C# will not have changed yet we'd want to update the coloring of the TagHelper to also colorize the HTML element. Let me know if you want to chat more offline.